widget/textinput.go

```diff
@@ -421,6 +421,10 @@ func (t *TextInput) doGoStart() {
 	t.caret.ResetBlinking()
 }
 
+func (t *TextInput) End() {
```

Comment:
So I like the idea of this, but I think maybe we should just set all the doXXX methods to DoXXX methods so they're exported instead of creating public wrappers.